### ISSUE TITLE
[mdatagen] Define featuregates in metadata.yaml

### ISSUE THREAD

**Issue by dmitryax** [open-telemetry/opentelemetry-collector#9860]
We can define feature gates in the `metadata.yaml` and generate code + documentation from it. We can use the following YAML structure with a docker stats feature gate as an example:

```
feature_gates:
  - id: receiver.dockerstats.useScraperV2
    stage: stable
    to_version: 0.74.0
    description: |
      When enabled, the receiver will use the function ScrapeV2 to collect metrics. 
      This allows each metric to be turned off/on via config. 
      The new metrics are slightly different to the legacy implementation.
    reference_url: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9794
```

**Comment by mx-psi** [comment]
Duplicate of open-telemetry/opentelemetry-collector-contrib/issues/21801? We can keep both open I guess, since mine is more specific about each component that would need changes

**Comment by dmitryax** [comment]
> Duplicate of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21801? We can keep both open I guess, since mine is more specific about each component that would need changes

Right. I wasn't aware of that issue, sorry. We can use that one for making use of this feature and this one for introducing the capability itself.

**Comment by lavishpal** [comment]
@dmitryax Could you assign this issue to me and it will be helpful if you provide guidance on how to get started?

**Comment by narcis96** [comment]
As discussed with @mx-psi. I am taking care of it

**Comment by corsairier** [comment]
Hi @dmitryax ! I noticed this issue has been open for a while. Is it still up for grabs?

**Comment by iblancasa** [comment]
I guess this can be closed https://github.com/open-telemetry/opentelemetry-collector/pull/14437
